docs/migration-guides/prometheus-migration.md

```diff
@@ -0,0 +1,328 @@
+
+// For testing purposes, You can fake above by creating a deploy with an emptyDir volume:
+//    kubectl run --image=nginx:latest --overrides '{"spec":{"template":{"spec":{ "volumes": [{"name": "data", "emptyDir": {}}]}}}}' nginx-tgt
+//    kubectl get deploy nginx-tgt -ojson > target.json
+
+local kube = import "./kube.libsonnet";
+local rsync_conf_b64 = std.base64(
+  |||
+    [data]
+    path = /data
+    list = false
+    use chroot = false
+    read only = true
+  |||
+);
+
+local rsync_container_name = "sidecar-rsync";
+
+// sidecar-rsync container (mounting <volume_name>) to be exposed via rsync
+local rsync_container = kube.Container(rsync_container_name) {
+  image: "alpine:3.8",
+  args: [
+    "sh",
+    "-c",
+    |||
+      apk update && apk add rsync curl
+      echo '%s' | base64 -d > /rsync.conf
+      rsync -vvv --daemon --no-detach --config=/rsync.conf
+    ||| % [rsync_conf_b64],
+  ],
+  volumeMounts_+: {
+    [volume_name]: { mountPath: "/data" },
+  },
+};
+
+local containers_obj(containers) = {
+  [container.name]: container for container in containers
+};
+
+local stack = {
+  
+  target_with_sidecar: target {
+    spec+: {
+      template+: {
+        spec+: {
+          // convert containers array to obj for easier jsonnet
+          // handling: in particular, if we just added the new container
+          // to existing containers[] array, we wouldn't be able to re-run
+          // this, as we'd get duplicated containers.
+          // By referring the added container via rsync_container_name, we
+          // can easily override it.
+
+          local containers = super.containers,
+          containers: kube.PodSpec {
+            default_container:: containers[0].name,
+            // create or override(existing) rsync_container
+            containers_+: containers_obj(containers) {
+              [rsync_container_name]: rsync_container,
+              ["prometheus"]+:{
+                args+: [
+                  "--web.enable-admin-api",
+                ],
+              },
+            },
+          }.containers,
+        },
+      },
+    },
+  },
+};
+
+kube.List() {items_+: stack}
+```
+
+Notice also how one extra argument has been added to the Prometheus container. In order to generate a snapshot of Prometheus its mandatory to have the admin api enabled. Take a look at the [Prometheus docs](https://prometheus.io/docs/prometheus/latest/querying/api/#tsdb-admin-apis) for more information about this.
+
+After deploying the generated manifest you will have a rsync container that you can access with:
+
+`kubectl exec -it -n NAMESPACE PROMETHEUS_POD_NAME -c sidecar-rsync sh`
+
+> Note: Make sure you change the placeholders NAMESPACE and PROMETHEUS_POD_NAME for the Kubernetes namespace and Prometheus pod name you have. In example: `kubectl exec -it -n monitoring prometheus-d776d68fb-vc7gt -c sidecar-rsync sh`
+
+Check also that a new service has been created for Prometheus:
```

Comment:
Check also that a new service has been created for Prometheus, as shown in the example command and output below:

Reply:
done